### DISCUSSION TITLE
Add config setting for changing "variant" keyword that is used as URL param in variant URL

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/commerce#2432] [Ideas]
This keyword is hardcoded in variant element class, here:

```
    public function getUrl(): string
    {
        return $this->product->url . '?variant=' . $this->id;
    }
```

**Comment by lukeholder** [comment]
Add a `variantUrlParameterName` column/field to Product Types table and product type editor page.

Description: The name of the Variant ID url parameter used when constructing a variant's URL
   
Default is `variant`. The Default variant URL would therefore be `<productUrl>?variant=<variantId>`.

Value would be inserted here: https://github.com/craftcms/commerce/blob/develop/src/elements/Variant.php#L564

**Comment by alexanderloewe** [comment]
1+ handle product slugs by commerce itself.
